fastly/ngwaf/v1/timeseries/api_test.go

```diff
@@ -0,0 +1,72 @@
+package timeseries
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/fastly/go-fastly/v10/fastly"
+)
+
+// Global value for the tests.
+var testWorkspaceID = fastly.TestNGWAFWorkspaceID
+
+const tsMetrics = "XSS,SQLI,HTTP404"
+
+var tsStart = time.Now().Add(-24 * time.Hour).UTC().Format(time.RFC3339)
+var tsEnd = time.Now().UTC().Format(time.RFC3339)
+var tsGranularity = 60
+
+func TestTime_Series(t *testing.T) {
+	t.Parallel()
+
+	var err error
+	var ts *TimeSeries
+
+	// Get request timeseries metrics for given workspace
+	fastly.Record(t, "get_timeseries", func(c *fastly.Client) {
+		ts, err = Get(c, &GetInput{
+			End:         &tsEnd,
+			Granularity: &tsGranularity,
+			Start:       &tsStart,
+			Metrics:     fastly.ToPointer(tsMetrics),
+			WorkspaceID: &testWorkspaceID,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
```

Comment:
```
 --- FAIL: TestTime_Series (0.00s)
    api_test.go:37: Get "https://api.fastly.com/ngwaf/v1/workspaces/`alk6DTsYKHKucJCOIavaJM`/timeseries?end=2025-06-05T19%3A42%3A20Z&granularity=60&metrics=XSS%2CSQLI%2CHTTP404&start=2025-06-04T19%3A42%3A20Z": Requested interaction not found
```

This doesn't work because every time you run the test you set a  new start and end! Therefore, the request params can never match the recordings again.

You need something like this:

```
	// NOTE: Update this to a recent time when regenerating the test fixtures,
	// otherwise the data may be outside of retention and an error will be
	// returned.
	end := time.Date(2024, 11, 26, 0, 0, 0, 0, time.UTC)
	start := end.Add(-2 * 24 * time.Hour)
```